parsers/rhea/rhea.go

```diff
@@ -0,0 +1,742 @@
+	IsTransport          bool     `json:"istransport" db:"istransport"`
+	Ec                   string   `json:"ec" db:"ec"`
+	Location             string   `json:"location" db:"location"`
+	Citations            []string `json:"citations"`
+	Substrates           []string `json:"substrates"`
+	Products             []string `json:"products"`
+	SubstrateOrProducts  []string `json:"substrateOrProducts"`
+}
+
+/******************************************************************************
+
+Parse functions
+
+These functions take in the rhea.rdf.gz dump file and return a Rhea struct,
+which contains all of the higher level structs
+
+******************************************************************************/
+
+// ParseRhea parses a list of bytes into a higher-level Rhea Struct.
+func ParseRhea(rheaBytes []byte) (Rhea, error) {
+	var err error
+	// Read rheaBytes into a Rdf object
+	var rdf Rdf
+	err = xml.Unmarshal(rheaBytes, &rdf)
+	if err != nil {
+		return Rhea{}, err
+	}
+
+	// Initialize Rhea
+	var rhea Rhea
+	compoundParticipantMap := make(map[string]string)
+	compoundMap := make(map[string]Compound)
+
+	for _, description := range rdf.Descriptions {
+		// Handle the case of a single compound -> reactive part, such as
+		// <rdf:Description rdf:about="http://rdf.rhea-db.org/Compound_10594">
+		// 	<rh:reactivePart rdf:resource="http://rdf.rhea-db.org/Compound_10594_rp2"/>
+		// </rdf:Description>
+		if (len(description.Subclass) == 0) && (description.ReactivePartXML.Resource != "") {
+			compoundParticipantMap[description.ReactivePartXML.Resource] = description.About
+		}
+		if description.Compound.Resource != "" {
+			compoundParticipantMap[description.About] = description.Compound.Resource
+		}
+
+		for _, subclass := range description.Subclass {
+			switch subclass.Resource {
+			case "http://rdf.rhea-db.org/DirectionalReaction":
+				newReaction := Reaction{
+					ID:                   description.ID,
+					Directional:          true,
+					Accession:            description.Accession,
+					Status:               description.Status.Resource,
+					Comment:              description.Comment,
+					Equation:             description.Equation,
+					HTMLEquation:         description.HTMLEquation,
+					IsChemicallyBalanced: description.IsChemicallyBalanced,
+					IsTransport:          description.IsTransport,
+					Ec:                   description.EC.Resource,
+					Citations:            description.CitationStrings(),
+					Substrates:           description.SubstrateStrings(),
+					Products:             description.ProductStrings(),
+					SubstrateOrProducts:  description.SubstrateOrProductStrings(),
+					Location:             description.Location.Resource}
+				rhea.Reactions = append(rhea.Reactions, newReaction)
+			case "http://rdf.rhea-db.org/BidirectionalReaction":
+				newReaction := Reaction{
+					ID:                   description.ID,
+					Directional:          false,
```

Comment:
```suggestion
					Directional:          subclass.Resource == "http://rdf.rhea-db.org/DirectionalReaction",
```
Would allow you to collapse these two cases and make it a bit easier to follow. Am I missing other differences?

Reply:
Yes! It would! thank you! Adding that in.